src/dataflow/merge/test.py

```diff
@@ -35,122 +35,132 @@ def extra_var_column_value():
+    
+    expected_concatenated_data = MuData({'mod1': data_sample1.mod['mod1'], 'mod2': data_sample2.mod['mod2']})
+    assert_annotation_objects_equal(concatenated_data, expected_concatenated_data)
 
-    assert set(concatenated_data.var_names) ==  (set(data_sample1.var_names) | set(data_sample2.var_names))
-    assert concatenated_data.var_keys() == ['gene_id', 'feature_type', 'genome']
 
-@pytest.mark.parametrize("mudata_non_overlapping_observations", [input_sample1_file], indirect=["mudata_non_overlapping_observations"])
-def test_merge_non_overlapping_observations(run_component, mudata_non_overlapping_observations):
+@pytest.mark.parametrize("mudata_non_overlapping_observations", ["small_mudata_mod1_path"], indirect=["mudata_non_overlapping_observations"])
+def test_merge_non_overlapping_observations(run_component, mudata_non_overlapping_observations, random_h5mu_path, small_mudata_mod2_path):
     """
     Merge with differing observations in the samples
     """
-    edited_h5mu, removed_observation_name = mudata_non_overlapping_observations
+    edited_h5mu_path, removed_observation_name = mudata_non_overlapping_observations
+    output_path = random_h5mu_path()
     # Remove 1 observation
     run_component([
-        "--input", edited_h5mu,
-        "--input", input_sample2_file,
-        "--output", "merge.h5mu"])
-    assert  Path("merge.h5mu").is_file()
-    concatenated_data = md.read("merge.h5mu", backed=False)
-    data_sample1 = md.read(edited_h5mu, backed=False)
-    data_sample2 = md.read(input_sample2_file, backed=False)
-    assert concatenated_data[removed_observation_name:]['rna'].n_obs == 0
-    assert concatenated_data[removed_observation_name:]['prot'].n_obs == 1
-
-    assert set(concatenated_data.obs_names) == (set(data_sample1.obs_names) | set(data_sample2.obs_names))
-    np.testing.assert_equal(concatenated_data[removed_observation_name:]['rna'].X.data,
-                            np.array([]))
-    np.testing.assert_equal(concatenated_data.copy()[removed_observation_name:]['prot'].X.data,
-                            data_sample2.copy()[removed_observation_name:]['prot'].X.data)
+        "--input", edited_h5mu_path,
+        "--input", small_mudata_mod2_path,
+        "--output", output_path])
+    
+    assert  output_path.is_file()
+    concatenated_data = read_h5mu(output_path, backed=False)
+    data_sample1 = read_h5mu(edited_h5mu_path, backed=False)
+    data_sample2 = read_h5mu(small_mudata_mod2_path, backed=False)
     
-@pytest.mark.parametrize("extra_var_column_value,expected", [("bar", "bar"), (True, True), (0.1, 0.1), (np.nan, pd.NA)])
+    expected_concatenated_data = MuData({'mod1': data_sample1.mod['mod1'], 'mod2': data_sample2.mod['mod2']})
+    
+    assert concatenated_data[removed_observation_name:]['mod1'].n_obs == 0
+    assert concatenated_data[removed_observation_name:]['mod2'].n_obs == 1
+    assert_annotation_objects_equal(concatenated_data, expected_concatenated_data)
```

Comment:
Could you leave the old asserts in wrt the `obs_names`? I changed my mind a bit :)
Also, 
```
np.testing.assert_equal(concatenated_data.copy()[removed_observation_name:]['prot'].X.data,
                            data_sample2.copy()[removed_observation_name:]['prot'].X.data)
```
Checked the data from `removed_observation_name`, not just if there was one observation, so could you do something similar?

Reply:
Just put 

```
np.testing.assert_equal(concatenated_data.copy()[removed_observation_name:]['prot'].X.data,
                            data_sample2.copy()[removed_observation_name:]['prot'].X.data)
```

back in